### PR TITLE
Type object pattern #555

### PR DESCRIPTION
**Pull request by AnaghaSasikumar**
Type object pattern implementation

### CODE REVIEW

File: typeobjectpattern/src/main/java/com/iluwatar/typeobject/CandyGame.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  void handleChange(int points) {
+    System.out.println("+" + points + " points!");
+    this.totalPoints += points; 
+    printGameStatus();
+  }
+  
+  void round(int timeSoFar, int totalTime) {
+    long start = System.currentTimeMillis();
+    long end = System.currentTimeMillis();
+    while (end - start + timeSoFar < totalTime && continueRound()) {
```
Comment (IAmPramod) [inline typeobjectpattern/src/main/java/com/iluwatar/typeobject/CandyGame.java]: same here. `start` and `end` are essentially the same.

Comment (AnaghaSasikumar) [reply typeobjectpattern/src/main/java/com/iluwatar/typeobject/CandyGame.java]: It's needed so that we can enter into while loop

**Comment by dfparker2002** [comment]
After reading the game article that inspired this addition, I would want to see how this pattern contribution is useful. May we see an effective extension of the breed-monster idea as candy.

How would we deliver the promise of the pattern; for example, overlay a rootbeer gummy into the base candy set to dynamically grow the candy set (monster) game?

**Comment by AnaghaSasikumar** [comment]
I am not sure what you mean by 'how', so do let me know if this doesn't answer your question, but I think it's a change you could probably make in the json file, assuming that we only want some kind of change in the property of the candy and not some behavioural change, i.e. maybe having this overlay would make it have more points or something like that. What I understood from the pattern was that it makes that kind of thing easier, but there will be some trade-off when it comes to behaioural change, which can't be put in the json file. It is a pretty simple pattern, I was just attracted to the idea of being able to implement so many different "types" of object using a single class and just a json file.

**Comment by iluwatar** [comment]
Good job @AnaghaSasikumar :+1:

**Pull request merged**